scipy/spatial/transform/_rotation.pyx

```diff
@@ -1718,6 +1984,172 @@ cdef class Rotation:
         """
         return self._compute_euler(seq, degrees, 'from_quat')
 
+    @cython.embedsignature(True)
+    def as_davenport(self, axes, extrinsic=True, degrees=False, correct_set=False):
+        """Represent as Davenport angles.
+
+        Any orientation can be expressed as a composition of 3 elementary
+        rotations.
+
+        For Euler angles, we have the following conditions:
+
+            - ``dot(axis1, axis2) = 0``
+            - ``dot(axis2, axis3) = 0``
+            - ``dot(axis1, axis3) = -1``, ``0`` or ``1``
+
+        For Davenport angles, the third relation is relaxed [1]_:
+            - ``dot(axis1, axis3)`` can be any value between -1 and 1.
+
+        A slightly modified version of the algorithm from [2]_ has been used to
+        calculate Davenport angles for the rotation about a given sequence of
+        axes.
+
+        Davenport angles, just like Euler angles, suffer from the problem of
+        gimbal lock [3]_, where the representation loses a degree of freedom
+        and it is not possible to determine the first and third angles
+        uniquely. In this case, a warning is raised, and the third angle is set
+        to zero. Note however that the returned angles still represent the
+        correct rotation.
+
+        Parameters
+        ----------
+        axes : list of (3, 1) arrays
+            Specifies sequence of axes for rotations. Must be a list of
+            3 arrays of shape (3, 1).
+        extrinsic : boolean, optional
+            If True, sequence will be extrinsic. If False, sequence will be
+            treated as intrinsic. Default is True.
+        degrees : boolean, optional
+            Returned angles are in degrees if this flag is True, else they are
+            in radians. Default is False.
+        correct_set : boolean, optional
+            If True, the angles are set so they match the Euler angles implementation.
+            Default is False.
+
+        Returns
+        -------
+        angles : ndarray, shape (3,) or (N, 3)
+            Shape depends on shape of inputs used to initialize object.
+            The returned angles are in the range:
+
+            - First angle belongs to [-180, 180] degrees (both inclusive)
+            - Third angle belongs to [-180, 180] degrees (both inclusive)
+            - Second angle belongs to a set of size 180 degrees.
+
+            If ``correct_set=False``, the set is ``[-lambda, 180 - lambda]``
```

Comment:
Well I think it's clear that this level of details is unnecessary and `correct_set=True` should implicit default.

Are there any advantages of `correct_set=False` which I'm missing?

Reply:
It is just an alternative solution, can be completely removed.

Reply:
For example, some methods in the `Eigen` project did not take enough care with this and gave confusing results, but in order not to break backwards compatibility they decided to leave it.

If you find this interesting, I could improve both `as_euler` and `as_davenport` to give either the `canonical` or `non_canonical` solutions. I have deduced a direct formula for this and it's pretty straightforward.

Reply:
Does `as_euler` works in "canonical" mode now? I think it's reasonable to stick to this convention in both `as_euler` and `as_davenport`. If necessary, a flag to return an alternative set can be added later (I doubt it's necessary though).

Reply:
It does work in "canonical" mode, indeed.